Added Lesson08 Circle.py and Revised mailroom part 2 and mailroom part 3

Chieu1 opened the pull request.

natasha-aleksandrova commented on students/chieu_quach/lesson08/circle.py:
```diff
@@ -0,0 +1,159 @@
+        return self.radius * 2   
+
+    @get_diameter.setter
+    def get_diameter(self, value):
```
note that in directions this was called `diameter` (properties are usually nouns)

natasha-aleksandrova commented on students/chieu_quach/lesson08/circle.py:
```diff
@@ -0,0 +1,159 @@
+        return repr(self.radius)
+
+    def __str__(self):
+        return str(self.radius)
```
the idea with `__str__` and `__repr__` is that you customize what gets printed out

natasha-aleksandrova commented on students/chieu_quach/lesson08/circle.py:
```diff
@@ -0,0 +1,159 @@
+
+    
+    
+    name = property(get_diameter, area)
```
Step 5 is missing? `from_diameter` alternate constructor?

natasha-aleksandrova commented on students/chieu_quach/lesson08/circle.py:
```diff
@@ -0,0 +1,159 @@
+
+    def __rmul__(self):
+        
+        return self.__mul__(other)
```
where is `other` coming from?

natasha-aleksandrova commented on students/chieu_quach/lesson08/circle_test.py:
```diff
@@ -0,0 +1,79 @@
+def test_mult():
+    c = Circle(2)
+    num = 3
+    assert repr(c * num) == '6'
```
your test should assert on class attribute values instead of using `repr()` method.

natasha-aleksandrova commented:
@Chieu1 I'm unable to merge due to merge conflicts. please make sure to merge with upstream and then we can see if it resolves the issues.

Chieu1 commented:
Natasha,

Per your comment, I have updated Circles exercise from lesson 8 to include step 5. I did PR on circle exercise for you to review. Please let me know if you see it.

For the mailroom 2 and 3 exercises,
I have corrected the indentation and resubmitted earlier. Can you let me know if you see them ? 
Previously,  I got 13 out of 15 for those two exercises because they have indentation violation.

Thanks,

Chieu

Sent from my iPhone

> On Jun 15, 2019, at 8:45 PM, Natasha <notifications@github.com> wrote:
> 
> @Chieu1 I'm unable to merge due to merge conflicts. please make sure to merge with upstream and then we can see if it resolves the issues.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

natasha-aleksandrova commented on students/chieu_quach/lesson08/circle.py:
```diff
@@ -0,0 +1,168 @@
+    def from_diameter(cls, radius):
+        cls.radius = radius / 2
+        
+        return cls(cls.radius)
```
this is pretty close...


should be:
```
@classmethod
    def from_diameter(cls, diameter):
        return cls(diameter / 2.0)
```

natasha-aleksandrova commented:
per my comment before, I cannot merge this due to conflicts - you need to merge with upstream.

Chieu1 commented:
Natasha 

Thank you very much for your help.

You are a very good instructor. 

I remembered that I had problem with the sort method in mailroom exercise because I used dictionary of dictionaries method. I could not sort in ascendind order. I called you at Zoom meeting and you told me to re-work on the dictionary. At that time, I was mad at you.  I am really sorry about that. I did what you suggested and was able to sort it in ascending order. If you had not provided me with that suggestion, I don’t think I would complete mailrooms 3 and 4 and pass this course.

Again, thank you very much for your help!

Appreciate your help.

I will try to complete lesson 9 and submit for review by either Wednesday ir Thursday

Thanks,

Chieu



Sent from my iPhone

> On Jun 17, 2019, at 8:57 AM, Natasha <notifications@github.com> wrote:
> 
> @natasha-aleksandrova commented on this pull request.
> 
> @Chieu1 adjusted your lesson 8 and mailroom grade
> 
> FYI you now have at least 80 percent for passing.
> 
> In students/chieu_quach/lesson08/circle.py:
> 
> > +
> +    @property
> +    def get_diameter(self):
> +   # def get_diameter(self, value):
> +      
> +        return self.radius * 2   
> +
> +    @get_diameter.setter
> +    def get_diameter(self, value):
> +        self.radius = value
> +
> +    @classmethod
> +    def from_diameter(cls, radius):
> +        cls.radius = radius / 2
> +        
> +        return cls(cls.radius)
> this is pretty close...
> 
> should be:
> 
> @classmethod
>     def from_diameter(cls, diameter):
>         return cls(diameter / 2.0)
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

natasha-aleksandrova commented:
I'm really happy you were able to get it working!

Chieu1 commented:
Natasha and Charles,

It has been my pleasure to have you as my Instructor and Tutor for this python course. 

I’ve learned  a lot from this challenging course. At time I thought I would not complete and pass this course. With your help to guide me through it and to allow me to have two weeks extended time, I am able to complete and pass this course.

Thanks Natasha for taking your time to provide comments on my gradings and to assist me with my homework at Zoom meeting.

Thanks Charles to assist me with my homework and to provide comments on my gradings.

Please help me to answer this question, Will I receive anything after I pass this course ? I need to turn in something to my company HR in order to get  tuition reimbursement from them.

Again, thank you very much Natasha and Charles 

Chieu

Sent from my iPhone

> On Jun 18, 2019, at 8:48 AM, Natasha <notifications@github.com> wrote:
> 
> I'm really happy you were able to get it working!
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

natasha-aleksandrova commented:
@Chieu1 send us an email and we can provide additional information to you privately.

Chieu1 commented:
Natasha,

This is my email chieu.quach@yahoo.com

Thank you very much for your help.

It was nice to have you as my Instructor. I hope to apply my learning skill in python in the workforce.

Thanks,

Chieu



Sent from my iPhone

> On Jun 21, 2019, at 4:11 PM, Natasha <notifications@github.com> wrote:
> 
> @Chieu1 send us an email and we can provide additional information to you privately.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

natasha-aleksandrova commented:
@Chieu1 just to clarify - send _us_ an email message with your request. This is a public space.

Chieu1 commented:
Natasha,

I am requesting that you send me information about my passing grade in python course. Please send me information including paid tuition fee, course name, and passing grade comfirmation

Thanks

Chieu

Sent from my iPhone

> On Jun 21, 2019, at 6:40 PM, Natasha <notifications@github.com> wrote:
> 
> @Chieu1 just to clarify - send us an email message with your request. This is a public space.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

Chieu1 commented:
Natasha,

I will send email message to the University to request for paid tuition fee and python passing grade info.

Thanks,

Chieu

Sent from my iPhone

> On Jun 21, 2019, at 6:40 PM, Natasha <notifications@github.com> wrote:
> 
> @Chieu1 just to clarify - send us an email message with your request. This is a public space.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
